### PR TITLE
fix(qprofiler): don't assume type of SAM attribute

### PR DESCRIPTION
**Pull request by holmeso**
# Description

Updated `qprofiler` so that it will only attempt to cast/parse SAM attribute into a particular type if it is an `instanceof` that type.

Fixes # 353

## Type of change

Please delete options that are not relevant.

- [X] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Unit test has been added and code has been run against BAM file that produced the error originally (successfully)

# Are WDL Updates Required?

no

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by holmeso** [comment]
Apologies - this is hard to review because of the whole file formatting that I did. Key lines of BamSummaryReport file are 579-588, and 629-645.

File: qprofiler/src/org/qcmg/qprofiler/bam/BamSummaryReport.java [hunk 1]
```diff
@@ -55,802 +59,803 @@
...
+            }
+            // CS
+            value = (String) record.getAttribute(SAMTag.CS);
+            if (null != value) {
+                SummaryByCycleUtils.parseCharacterSummary(tagCSByCycle, value, null, 1);
+                SummaryReportUtils.tallyBadReadsAsString(value, csBadReadLineLengths);
+            }
+        }
+
+        parseStringAttribute(record.getAttribute("RG"), tagRGLineLengths);
```
Comment (delocalizer) [inline qprofiler/src/org/qcmg/qprofiler/bam/BamSummaryReport.java:579]: I understand that this `parse*` naming is consistent with the usage already established in this file but would you consider using instead  `aggregate*` or `accumulate*` for these two new methods.

**Review by delocalizer — APPROVED** [review]
Looks good, thanks for the fix!

File: qprofiler/src/org/qcmg/qprofiler/bam/BamSummaryReport.java (hunk 1 above)
Comment (holmeso) [reply qprofiler/src/org/qcmg/qprofiler/bam/BamSummaryReport.java]: yep - good idea

**Pull request merged**